test/libsolidity/ASTPropertyTest.cpp

```diff
@@ -110,38 +110,40 @@ void ASTPropertyTest::readTestedProperties(Json::Value const& _astJson)
 {
 	queue<Json::Value> nodesToVisit;
 	nodesToVisit.push(_astJson);
-	string documentation = "documentation";
-	string testCaseLine;
 
-	while(!nodesToVisit.empty())
+	while (!nodesToVisit.empty())
 	{
-		auto& node = nodesToVisit.front();
-		if (node.isObject())
-		{
-			for (auto memberName: node.getMemberNames())
-				if (memberName == documentation)
-				{
-					auto docNode = node[documentation];
-					testCaseLine = docNode.isObject() ?
-						docNode["text"].asString() :
-						docNode.asString();
-
-					soltestAssert(!testCaseLine.empty());
-					auto pairs = readKeyValuePairs(testCaseLine);
-					soltestAssert(!pairs.empty());
-					auto [testId, testedProperty] = pairs[0];
-					m_testCases[testId].property = testedProperty;
-					auto propertyNode = findNode(node, testedProperty);
-					soltestAssert(propertyNode.has_value(), "Could not find "s + testedProperty);
-					soltestAssert(!propertyNode->isObject());
-					m_testCases[testId].obtainedValue = propertyNode->asString();
-				}
-				else
-					nodesToVisit.push(node[memberName]);
-		}
-		else if (node.isArray())
+		Json::Value& node = nodesToVisit.front();
+
+		if (node.isArray())
 			for (auto&& member: node)
 				nodesToVisit.push(member);
+		else if (node.isObject())
+			for (string const& memberName: node.getMemberNames())
+			{
+				if (memberName != "documentation")
+				{
+					nodesToVisit.push(node[memberName]);
+					continue;
+				}
+
+				string nodeDocstring = node["documentation"].isObject() ?
+					node["documentation"]["text"].asString() :
+					node["documentation"].asString();
+				soltestAssert(!nodeDocstring.empty());
+
+				vector<StringPair> pairs = readKeyValuePairs(nodeDocstring);
+				soltestAssert(!pairs.empty());
+
+				auto const& [testId, testedProperty] = pairs[0];
+				m_testCases[testId].property = testedProperty;
+
+				soltestAssert(node.isMember("nodeType"));
```

Comment:
I guess that this prevent that any metadata (?) node or something which is not AST node to be processed, but I am not sure if we need an informative error message here and how to phrase.

Reply:
I did not add one here because if it fails, it's really a problem with the AST and not with the test. So it's instead something that would be a part of expectations if you care about the value of this field. This assert is here only because I included node type in the other assert below. Other that that, we would have been fine not checking for it.

But looking at it now, it would not hurt to have a message anyway. After all we're testing the AST and you could theoretically get a malformed one if you're changing code responsible for generating it. I would not expect this to happen very often, but still. You could use something like `"Invalid AST node: 'nodeType' field is missing."`.